src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
I used `grep 'argt\>' src/nvim/* -r` to look for `(struct exarg).argt`. There are only a few places where an assignment is used (`=`,`|=`). Only `uint32_t` values are assigned, so I think it would be easier to just change the type from `long` to `uint32_t` for `(struct exarg).argt`.